### PR TITLE
Set up Jest

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

File: jujugui/static/gui/src/app/components/help/test-help.js [hunk 1]
```diff
@@ -22,132 +20,7 @@ describe('Help', () => {
...
-                target="_blank">
-                The Juju project
-              </a>
-            </p>
-          </div>
-        </footer>
-      </div>
-    );
-    assert.compareJSX(wrapper.find('.help__body'), expected);
+    expect(wrapper).toMatchSnapshot();
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/help/test-help.js]: I've updated this one component as an example.

Comment (frankban) [reply jujugui/static/gui/src/app/components/help/test-help.js]: Nice

Comment (barrymcgee) [reply jujugui/static/gui/src/app/components/help/test-help.js]: 💥🙌🏻💪🏻

File: .babelrc [hunk 2]
```diff
@@ -13,6 +13,15 @@
       "plugins": [
         "rewire"
```
Comment (hatched) [inline .babelrc]: Do we still need rewire in development if it's in the test?

File: Makefile [hunk 3]
```diff
@@ -293,6 +295,18 @@ test-js-old: gui
...
 
+.PHONY: test-jest
+test-jest: gui
+	$(JEST)
+
+.PHONY: watch-jest
+watch-jest: gui
+	$(JEST) --watchAll
+
+.PHONY: watch-jest
```
Comment (hatched) [inline Makefile]: This phony label is incorrect.

File: jujugui/static/gui/src/app/components/help/test-help.js (hunk 1 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/help/test-help.js]: 😆

File: package.json [hunk 4]
```diff
@@ -64,10 +68,12 @@
     "keysim": "2.0.1",
     "mkdirp": "0.5.1",
     "mocha": "5.1.1",
+    "mutation-observer": "^1.0.3",
```
Comment (hatched) [inline package.json]: I see a number of these are being added but not used? Can you give me an idea of why these are required?

**Comment by huwshimi** [comment]
[Set up Jest](https://trello.com/c/SLw5xjIy/872-set-up-jest)

File: .babelrc (hunk 2 above)
Comment (huwshimi) [reply .babelrc]: Unfortunately we rewire in files that are tested by Karma and Jest and Karma uses "development" and Jest uses "test".

File: jujugui/static/gui/src/test/jest-setup.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+global.assert = require('chai').assert;
+global.sinon = require('sinon');
+global.enzyme = require('enzyme');
+global.MutationObserver = require('mutation-observer');
```
Comment (huwshimi) [inline jujugui/static/gui/src/test/jest-setup.js]: This is where the mutation observer gets attached.

File: package.json (hunk 4 above)
Comment (huwshimi) [reply package.json]: This gets attached to the global test object as jsdom doesn't have a MutationObserver implementation by default. I've added a comment to this PR for where this happens.

**Comment by huwshimi** [comment]
Thanks everyone :shipit:

**Comment by huwshimi** [comment]
Oops, not the squirrel!

**Pull request merged**